python/pyiceberg/table/metadata.py

```diff
@@ -128,13 +138,13 @@ def construct_refs(cls, data: Dict[str, Any]):
     schemas: List[Schema] = Field(default_factory=list)
     """A list of schemas, stored as objects with schema-id."""
 
-    current_schema_id: int = Field(alias="current-schema-id", default=DEFAULT_SCHEMA_ID)
+    current_schema_id: int = Field(alias="current-schema-id", default=INITIAL_SCHEMA_ID)
```

Comment:
These probably shouldn't have defaults because they need to be explicitly set to some ID that exists in the list of schemas, specs, or sort orders.

Reply:
https://github.com/apache/iceberg/pull/5672#discussion_r962363711

Reply:
I think that defaulting the ID when creating a new schema, spec, or order is fine. But I don't think it is a good idea to default it here. At this point, we no longer have users constructing metadata by hand and we want to make sure that we're setting the ID correctly. If we re-create a schema for a new table metadata object, then we should also set the current schema ID to that schema's ID rather than relying on the same default in two places. That way if we ever change the default assignment we don't break tables.

Reply:
Fair point. I've removed this set explicitly when we get a v1 metadata.